aodn_cloud_optimised/bin/generic_cloud_optimised_creation.py

```diff
@@ -49,131 +27,236 @@
 from aodn_cloud_optimised.lib.s3Tools import s3_ls
 
 
+# ---------- Pydantic schema for validating the config ----------
+class ClusterConfig(BaseModel):
+    mode: Optional[str] = None
+    restart_every_path: bool = False
+
+
+class PathConfig(BaseModel):
+    s3_uri: str
+    filter: Optional[List[str]] = Field(default_factory=list)
+    year_range: Optional[List[int]] = None
+
+    @field_validator("year_range")
+    def validate_year_range(cls, v: Optional[List[int]]) -> Optional[List[int]]:
+        if v is None or len(v) == 0:
+            return None  # No year filtering
+
+        if not (1 <= len(v) <= 2):
+            raise ValueError("year_range must contain 1 or 2 integers")
+
+        # If one year, return as single-item list
+        if len(v) == 1:
+            return v
+
+        # If two years, return inclusive range list
+        start, end = v
+        if start > end:
+            raise ValueError("year_range start year must be <= end year")
+
+        return list(range(start, end + 1))
+
+    @field_validator("filter")
+    def validate_regex(cls, v):
+        for pattern in v:
+            try:
+                re.compile(pattern)
+            except re.error as e:
+                raise ValueError(f"Invalid regex: {pattern} ({e})")
+        return v
+
+
+class RunSettings(BaseModel):
+    paths: List[PathConfig]
+    cluster: ClusterConfig = Field(default_factory=ClusterConfig)
```

Comment:
```suggestion
    cluster: ClusterConfig = Field(description="The cluster config configuring how x and y etc...")
```

Should this have a default? I would assume you need to supply this particularly and a default option won't do. Could be wrong tho